File: src/Backend/Domain/Auth.Entity/Ids4Entity/ApiScope.cs
using System;
using Dapper.Contrib.Plus;

namespace Auth.Entity.Ids4Entity
{
    /// <summary>
    /// ApiScope
    /// </summary>
    [Table("api_scopes")]
    public class ApiScope
    {
        /// <summary>
        /// 主键
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Enabled
        /// </summary>
        public bool Enabled { get; set; }

        /// <summary>
        /// Name
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public string DisplayName { get; set; }

        /// <summary>
        /// Description
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Required
        /// </summary>
        public bool Required { get; set; }

        /// <summary>
        /// Emphasize
        /// </summary>
        public bool Emphasize { get; set; }

        /// <summary>
        /// ShowInDiscoveryDocument
        /// </summary>
        public object ShowInDiscoveryDocument { get; set; }

        /// <summary>
        /// ApiScopeClaim
        /// </summary>
        [Computed]
        public ApiScopeClaim ApiScopeClaim { get; set; }

        /// <summary>
        /// ApiScopeProperty
        /// </summary>
        [Computed]
        public ApiScopeProperty ApiScopeProperty { get; set; }
    }
}

File: src/FrontEnd/auth-admin/src/types/entities/ids/api-scope.ts
interface IApiScope {
  id?: string;
  enabled: boolean;
  name: string;
  displayName: string;
  description: string;
  required: boolean;
  emphasize: boolean;
  showInDiscoveryDocument: any;
}

export default class ApiScope implements IApiScope {
  id?: string;
  enabled: boolean = true;
  name: string = '';
  displayName: string = '';
  description: string = '';
  required: boolean = false;
  emphasize: boolean = false;
  showInDiscoveryDocument: any = null;
}